src/Notifier.js

```diff
@@ -133,29 +153,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
+                // Attempt to get permission from user
+                var _this = this;
+                global.Notification.requestPermission().then(function(result) {
```

Comment:
I don't think we can rely on our users having browsers which support the promise-returning version of `requestPermission`. https://developer.mozilla.org/en-US/docs/Web/API/Notification/requestPermission suggests that Chrome doesn't support it at all. You'll need to stick with the callback version.